CMakeLists.txt

```diff
@@ -11,7 +11,7 @@
 # or implied. See the License for the specific language governing permissions and limitations under
 # the License.
 # =============================================================================
-cmake_minimum_required(VERSION 3.18...3.18 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
```

Comment:
Is there a reason for this change? We use features that require 3.18.

Reply:
This PR is an exploration... definitely WIP